src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```
@@ -351,6 +358,68 @@ public async Task SendStreamCompletionAsync(string connectionId, string streamId
         await _restClient.SendWithRetryAsync(api, HttpMethod.Post, cancellationToken: cancellationToken);
     }
 
+#if NET7_0_OR_GREATER
+#nullable enable
+    public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object?[] args, CancellationToken cancellationToken = default)
+    {
+        // Validate input parameters
+        if (string.IsNullOrEmpty(methodName))
+        {
+            throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(methodName));
+        }
+        if (string.IsNullOrEmpty(connectionId))
+        {
+            throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+        }
+
+        // Get API endpoint and prepare for the request
+        var api = await _restApiProvider.SendClientInvocationAsync(_appName, _hubName, connectionId);
+        string? responseContent = null;
+        var isSuccessStatusCode = false;
+        // Send request and capture the response
+        await _restClient.SendMessageWithRetryAsync(
+            api,
+            HttpMethod.Post,
+            methodName,
+            args,
+            async response =>
+            {
+                responseContent = await response.Content.ReadAsStringAsync();
+                isSuccessStatusCode = response.IsSuccessStatusCode;
+                return true;
```

The return bool indicates whether you want the response to be further processed by `ThrowExceptionOnResponseFailureAsync` in this class. 404 not found, 401 unauthorized, are some of the errors that you want to throw a universe exception via `ThrowExceptionOnResponseFailureAsync`.

---

Updated to throwing on unexpected errors